### PR TITLE
[Navigation] Fixed visual regressions

### PR DESCRIPTION
**Pull request by tjonx**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-summer-editions/issues/987
Fixes https://github.com/Shopify/polaris-summer-editions/issues/1004

### WHAT is this pull request doing?

- Fixes various `SecondaryActions` visual regressions

https://github.com/Shopify/polaris/assets/67433661/63c6eee2-018b-49f4-9826-2d90f6ca0a99

### CODE REVIEW

File: polaris-react/src/components/Navigation/Navigation.scss [hunk 1]
```diff
@@ -582,6 +605,16 @@ $nav-max-width: 360px;
         opacity: 1;
       }
     }
+
+    &:active {
+      #{$se23} & {
+        // stylelint-disable-next-line -- se23
+        .SecondaryActions {
+          // stylelint-disable-next-line -- required for specific styling se23
+          background-color: var(--p-color-bg-app-selected) !important;
```
Comment (jesstelford) [inline polaris-react/src/components/Navigation/Navigation.scss]: I think this `!important` could be avoided by modifying the code at line 181:

https://github.com/Shopify/polaris/blob/cb7a0615fa5ad42697855d5ce387f3960eba51df/polaris-react/src/components/Navigation/Navigation.scss#L181-L202

To invert the pseudo & class selectors:

```scss
    &:is(:hover, :focus-visible) {
      background: var(--p-color-bg-subdued-hover);
    }

    // stylelint-disable selector-max-class, max-nesting-depth, selector-max-combinators -- se23
    &.ItemInnerWrapper-display-actions-on-hover {
      @media #{$p-breakpoints-md-up} {
        &:is(:hover, :focus-visible) {
          .SecondaryActions {
            background: var(--secondary-actions-on-hover-background-color);

            &::before {
              // stylelint-disable-next-line -- required for fade effect
              background: linear-gradient(
                to right,
                rgba(0, 0, 0, 0),
                var(--p-color-bg-subdued-hover) var(--p-space-8)
              );
            }
          }
        }
        
        &:is(:active) {
          .SecondaryActions {
            background: var(--p-color-bg-app-selected);
          }
        }
      }
      // stylelint-enable
    }
```

**Review by sarahill — APPROVED** [review]
✨ ✅

**Pull request merged**